File: nodecore-cli/src/main/java/nodecore/cli/serialization/SubmitMultisigTxPayload.java
// VeriBlock NodeCore CLI
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.cli.serialization;

import nodecore.api.grpc.VeriBlockMessages;
import nodecore.api.grpc.utilities.ByteStringUtility;

public class SubmitMultisigTxPayload {
    public SubmitMultisigTxPayload(final VeriBlockMessages.SubmitMultisigTxReply reply) {
        this.transaction = new TransactionInfo(reply.getSignedMultisigTransaction().getTransaction());
        this.multisigSlotBundle = new MultisigBundleInfo(reply.getSignedMultisigTransaction().getSignatureBundle());
        this.txid = ByteStringUtility.byteStringToHex(reply.getTxid());
        this.sigIndex = reply.getSignedMultisigTransaction().getSignatureIndex();
    }

    public TransactionInfo transaction;
    public String txid;
    public MultisigBundleInfo multisigSlotBundle;
    public long sigIndex;
}

File: nodecore-cli/src/main/java/nodecore/cli/serialization/SubmitMultisigTxPayload.kt
// VeriBlock NodeCore CLI
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.cli.serialization

import nodecore.api.grpc.VeriBlockMessages.SubmitMultisigTxReply
import nodecore.api.grpc.utilities.ByteStringUtility
import nodecore.api.grpc.utilities.extensions.toHex

class SubmitMultisigTxPayload(
    reply: SubmitMultisigTxReply
) {
    val transaction = TransactionInfo(reply.signedMultisigTransaction.transaction)

    val txid = reply.txid.toHex()

    val multisigSlotBundle = MultisigBundleInfo(reply.signedMultisigTransaction.signatureBundle)

    val sigIndex = reply.signedMultisigTransaction.signatureIndex
}
